code/datums/traits/positive_quirk.dm

```diff
@@ -139,7 +139,10 @@
 //Credit To Yowii/Yoworii/Yorii for a much more streamlined method of language library building
 
 /datum/quirk/multilingual/add()
-	if(!known_language)
+	var/client/C = GLOB.directory[ckey(quirk_holder.key)]
+	if(C?.prefs)
+		known_language = C.prefs.read_character_preference(/datum/preference/choiced/quirk/multilingual_language)
+	if(!known_language) // default to random
```

Comment:
do keep in mind that prefs don't actually ever come out as null, you'd need to specify a special preference named "Random" and make it the default, as well as disable randomization for this preference to make it random in this way.